### PR TITLE
Add location types to SIR

### PR DESCRIPTION
**Pull request by lukasm91**
## Technical Description

- Add location types to SIR
- Serialization / deserialization of location types in SIR
- Fix python SIR-printer (missing reduction over neighbor)
- Pass location types from SIR to IIR
- Remove unused information stored in the IIRBuilder
- Removes some unneeded `inline`

### CODE REVIEW

**Comment by lukasm91** [comment]
launch jenkins

File: dawn/src/dawn/Serialization/ASTSerializer.cpp [hunk 1]
```diff
@@ -113,6 +113,21 @@ void setField(dawn::proto::statements::Field* fieldProto, const sir::Field* fiel
...
+    protoLocationType = proto::statements::Field_LocationType_Cell;
+    break;
+  case dawn::ast::Expr::LocationType::Edges:
+    protoLocationType = proto::statements::Field_LocationType_Edge;
+    break;
+  case dawn::ast::Expr::LocationType::Vertices:
+    protoLocationType = proto::statements::Field_LocationType_Vertex;
+    break;
+  default:
+    dawn_unreachable("");
```
Comment (twicki) [inline dawn/src/dawn/Serialization/ASTSerializer.cpp]: can we have a meaningful string here?

File: dawn/src/dawn/Serialization/ASTSerializer.cpp [hunk 2]
```diff
@@ -481,6 +496,19 @@ std::shared_ptr<sir::Field> makeField(const proto::statements::Field& fieldProto
...
+    field->locationType = ast::Expr::LocationType::Cells;
+    break;
+  case proto::statements::Field_LocationType_Edge:
+    field->locationType = ast::Expr::LocationType::Edges;
+    break;
+  case proto::statements::Field_LocationType_Vertex:
+    field->locationType = ast::Expr::LocationType::Vertices;
+    break;
+  default:
+    dawn_unreachable("");
```
Comment (twicki) [inline dawn/src/dawn/Serialization/ASTSerializer.cpp]: can we have a meaningful string here?

**Comment by lukasm91** [comment]
launch jenkins

**Pull request merged**